src/cpu/probes/looppoint.hh

```diff
@@ -0,0 +1,30 @@
+#ifndef __CPU_PROBES_LOOPPOINT_HH__
+#define __CPU_PROBES_LOOPPOINT_HH__
+
+#include "cpu/base.hh"
+#include "cpu/probes/looppointmanager.hh"
+#include "params/LoopPoint.hh"
+#include "sim/probe/probe.hh"
+
+namespace gem5
+{
+
+class LoopPoint : public ProbeListenerObject
+{
+    public:
+        LoopPoint(const LoopPointParams &params);
+        virtual ~LoopPoint();
+        virtual void init();
+        virtual void regProbeListeners();
+        void check_pc(const Addr&);
+
+    private:
+        const std::vector<Addr> targetPC;
```

Comment:
I suggest making this an `unordered_map<Addr,unsigned>`. It should perform better and you won't have to use `check_pc`. Instead, you can set this up at initialization time (maybe through `init()`?)

Reply:
Actually, I think this should be an `unordered_set`